portia/portia.py

```
@@ -263,25 +302,55 @@ def plan(
 
         return plan
 
+    def _coerce_plan_inputs(
+        self, plan_inputs: list[PlanInput] | list[dict[str, str]] | list[str] | None
+    ) -> list[PlanInput] | None:
+        """Coerce plan inputs from any input type into a list of PlanInputs we use internally."""
+        if plan_inputs is None:
+            return None
+        if isinstance(plan_inputs, list):
+            to_return = []
+            for plan_input in plan_inputs:
+                if isinstance(plan_input, dict):
+                    if "name" not in plan_input:
+                        raise ValueError("Plan input must have a name and description")
+                    to_return.append(
+                        PlanInput(
+                            name=plan_input["name"],
+                            description=plan_input.get("description", None),
+                        )
+                    )
+                elif isinstance(plan_input, str):
+                    to_return.append(PlanInput(name=plan_input))
+                else:
+                    to_return.append(plan_input)
+            return to_return
+        return None
```

I wonder if we should raise here instead, because we've already captured if plan_inputs is None, so if they reach this point it means they've passed something along to this function and returning None here might return nothing silently and cause potential issues upstream, where a Raise TypeError("Expected list[PlanInput], ...") would fail faster and highlight issues.

---

Good point - now updated!